contracts/modules/v2/Assessment.sol

```diff
@@ -0,0 +1,397 @@
+  function pool() internal view returns (IPool) {
+    return IPool(getInternalContractAddress(ID.P1));
+  }
+
+  function getVoteCountOfAssessor(address assessor) external override view returns (uint) {
+    return votesOf[assessor].length;
+  }
+
+  function getClaimsCount() external override view returns (uint) {
+    return claims.length;
+  }
+
+  function getIncidentsCount() external override view returns (uint) {
+    return incidents.length;
+  }
+
+  /* === MUTATIVE FUNCTIONS ==== */
+
+  /**
+   *  Submits a claim for assessment
+   *
+   *  @dev This function requires an ETH submission fee. See: _getSubmissionFee()
+   *
+   *  @param coverId          Cover identifier
+   *  @param requestedAmount  The amount expected to be received at payout
+   *  @param withProof        When true, a ProofSubmitted event is emitted with ipfsProofHash.
+   *                          When false, no ProofSubmitted event is emitted to save gas if the
+   *                          cover wording doesn't enforce a proof of loss.
+   *  @param ipfsProofHash    The IPFS hash required for proof of loss. It is ignored if withProof
+   *                          is false
+   */
+  function submitClaim(
+    uint24 coverId,
+    uint96 requestedAmount,
+    bool withProof,
+    string calldata ipfsProofHash
+  ) external payable onlyMember {
+    AssessmentClaimsLib.submitClaim(
+      CONFIG,
+      coverId,
+      requestedAmount,
+      withProof,
+      ipfsProofHash,
+      claims
+    );
+
+  }
+
+  function submitIncident(
+    uint24 productId,
+    uint96 priceBefore,
+    uint32 date
+  ) external {
+    (
+      AffectedToken memory affectedToken,
+      Incident memory incident
+    ) = AssessmentIncidentsLib.getIncidentToSubmit(
+      CONFIG,
+      nxm,
+      memberRoles(),
+      productId,
+      priceBefore,
+      date
+    );
+
+    AssessmentIncidentsLib.saveIncident (
+      incident,
+      incidents,
+      affectedToken,
+      tokenAffectedByIncident,
+      incidentProponent
+    );
+  }
+
+  function depositStake (uint96 amount) external onlyMember {
+    Stake storage stake = stakeOf[msg.sender];
+    stake.amount += amount;
+    nxm.transferFrom(msg.sender, address(this), amount);
+  }
+
+  function withdrawReward (address user, uint104 untilIndex) external {
+    AssessmentVoteLib.withdrawReward(
+      CONFIG,
+      nxm,
+      user,
+      untilIndex,
+      stakeOf,
+      votesOf,
+      claims,
+      incidents
+    );
+  }
+
+  function withdrawStake (uint96 amount) external onlyMember {
+    AssessmentVoteLib.withdrawStake(CONFIG, nxm, stakeOf, votesOf, amount);
+  }
+
+  function redeemClaimPayout (uint104 id, address payable coverOwner) external {
+    AssessmentClaimsLib.redeemClaimPayout(
+      CONFIG,
+      pool(),
+      memberRoles(),
+      id,
+      coverOwner,
+      claims,
+      addressOfAsset
+    );
+  }
+
+  function redeemIncidentPayout (uint104 incidentId, uint32 coverId, uint payoutAmount) external {
```

Comment:
when do you use redeemClaimPayout vs redeemIncidentPayout as a user?

Reply:
Claim is for protocol cover and custodian cover, where you have to make individual claims. Incident is for yield token cover where people don't need to make a claim but just have an accepted incident from which they can redeem their payout.